app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java

```diff
@@ -209,6 +209,16 @@ public long getLastMessageID(long threadId) {
     }
   }
 
+  public long getLastMessageTimestamp(long threadId) {
+    String order     = MmsSmsColumns.NORMALIZED_DATE_SENT + " DESC";
+    String selection = MmsSmsColumns.THREAD_ID + " = " + threadId;
+
+    try (Cursor cursor = queryTables(PROJECTION, selection, order, "1")) {
+      cursor.moveToFirst();
```

Comment:
`moveToFirst` will return false if there are no rows, in which case we might want to return -1 instead of carrying on to `getColumnIndexOrThrow` because it'll throw, and the (single) call for this isn't caught so it'll crash, yeah?

Reply:
fixed